src/tests/tracing/runtimeeventsource/NativeRuntimeLevelKeywords.cs

```diff
@@ -0,0 +1,285 @@
+﻿//*
+#define GC_COLLECT
+//*/
+//*
+#define GC_FINALIZERS
+//*/
+//*
+#define GC_MEMORY_PRESSURE
+//*/
+/*
+#define LOCK_CONTENTION //BUG!? Possibly deadlocks Linux x64
+//*/
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+#if LOCK_CONTENTION
+using System.Threading;
+using System.Threading.Tasks;
+#endif //LOCK_CONTENTION
+
+using System.Buffers; // Copied from https://github.com/dotnet/runtime/pull/86370 by @davmason:
+{
+    // Access ArrayPool.Shared.Rent() before the test to avoid the deadlock reported
+    // in https://github.com/dotnet/runtime/issues/86233. This is a real issue,
+    // but only seen if you have a short lived EventListener and create EventSources
+    // in your OnEventWritten callback so we don't expect customers to hit it.
+    byte[] localBuffer = ArrayPool<byte>.Shared.Rent(10);
+    Console.WriteLine($"buffer length={localBuffer.Length}");
+}
+
+const EventKeywords keywords = EventKeywords.None
+#if GC_COLLECT || GC_FINALIZERS || GC_MEMORY_PRESSURE
```

Comment:
I'm not sure I understand what you're trying to accomplish with the `#if` here. I don't see anywhere that defines these constants at build time, so they should always be false. I would expect the keywords to always be EventKeywords.None with the way this is laid out

Reply:
Oh, I didn't see the `#define` above. It would be better to not use `#if` unless we absolutely have to, it makes the test hard to follow.